server.rb

```diff
@@ -93,9 +93,15 @@ def html(view)
   end
 
   def sort_versions(ary)
-    valid_version_numbers = ary.select { |i| i if i.match(/(0-9|\.|\-)/)}
-    non_valid_version_numbers = ary - valid_version_numbers
-    (valid_version_numbers.sort_by {|v| Gem::Version.new( v.gsub(/^[a-z|A-Z|.]*/, '') ) } + non_valid_version_numbers)
+    valid_version_numbers = ary.select { |i| i if i.match(/^[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+(-[[:alnum:]]+)?$/) }
+    non_valid_version_numbers = (ary - valid_version_numbers).sort
+    versions = valid_version_numbers.sort_by {|v| Gem::Version.new( v.gsub(/^[a-z|A-Z|.]*/, '') ) } + non_valid_version_numbers
+    if versions.include?('latest')
```

Comment:
I don't think you need this block or the versions.reverse.. there's a reverse happening on the return of this method later in the script...

The goal here is to sort the versions with the newer at the top. 

As is here:

<img width="329" alt="crane_operator" src="https://cloud.githubusercontent.com/assets/428/26002061/0484c4ee-36f4-11e7-9c01-4dd7776de2b7.png">


Without this or the versions.sort... 

<img width="278" alt="crane_operator" src="https://cloud.githubusercontent.com/assets/428/26002081/1892bf86-36f4-11e7-92e5-1c0a986d2321.png">

The second one seems more correct to me.